### ISSUE TITLE
Improve style_transfer.ipynb: modularization, device handling, and optimization

### ISSUE THREAD

**Issue by tanmaytyagii** [tanmaytyagii/GenAI-Learning#1]
## Description
The current implementation of `style_transfer.ipynb` works but can be improved in terms of structure, reusability, and performance.

## Issues Identified
- Code is written in a monolithic notebook format without modular functions
- No explicit GPU/CPU device handling (e.g., `torch.device`)
- Repeated code blocks for preprocessing and transformations
- Lack of parameterization for hyperparameters (style weight, content weight, learning rate)
- Missing comments and documentation for key steps (loss functions, optimization loop)

## Proposed Improvements
- Refactor code into reusable functions (e.g., `load_image`, `compute_loss`, `train_step`)
- Add device handling for GPU acceleration:
  ```python
  device = torch.device("cuda" if torch.cuda.is_available() else "cpu")

**Comment by tanmaytyagii** [comment]
Issue acknowledged. Improvements will be handled in upcoming updates.
Closing for now.
